pkg/resources/authentication_policy.go

```diff
@@ -0,0 +1,388 @@
+	}); err == nil {
+		mfaAuthenticationMethods = append(mfaAuthenticationMethods, sdk.ParseCommaSeparatedStringArray(mfaAuthenticationMethodsProperty.Value, false)...)
+	}
+	if err = d.Set("mfa_authentication_methods", mfaAuthenticationMethods); err != nil {
+		return diag.FromErr(err)
+	}
+
+	mfaEnrollment, err := collections.FindFirst(authenticationPolicyDescriptions, func(prop sdk.AuthenticationPolicyDescription) bool { return prop.Property == "MFA_ENROLLMENT" })
+	if err == nil {
+		if err = d.Set("mfa_enrollment", mfaEnrollment.Value); err != nil {
+			return diag.FromErr(err)
+		}
+	}
+
+	clientTypes := make([]string, 0)
+	if clientTypesProperty, err := collections.FindFirst(authenticationPolicyDescriptions, func(prop sdk.AuthenticationPolicyDescription) bool { return prop.Property == "CLIENT_TYPES" }); err == nil {
+		clientTypes = append(clientTypes, sdk.ParseCommaSeparatedStringArray(clientTypesProperty.Value, false)...)
+	}
+	if err = d.Set("client_types", clientTypes); err != nil {
+		return diag.FromErr(err)
+	}
+
+	securityIntegrations := make([]string, 0)
+	if securityIntegrationsProperty, err := collections.FindFirst(authenticationPolicyDescriptions, func(prop sdk.AuthenticationPolicyDescription) bool { return prop.Property == "SECURITY_INTEGRATIONS" }); err == nil {
+		securityIntegrations = append(securityIntegrations, sdk.ParseCommaSeparatedStringArray(securityIntegrationsProperty.Value, false)...)
+	}
+	if err = d.Set("security_integrations", securityIntegrations); err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err = d.Set("comment", authenticationPolicy.Comment); err != nil {
+		return diag.FromErr(err)
+	}
+	if err := d.Set(FullyQualifiedNameAttributeName, id.FullyQualifiedName()); err != nil {
+		return diag.FromErr(err)
+	}
+
+	return diags
+}
+
+func UpdateContextAuthenticationPolicy(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*provider.Context).Client
+	id := helpers.DecodeSnowflakeID(d.Id()).(sdk.SchemaObjectIdentifier)
+
+	// change to authentication methods
+	if d.HasChange("authentication_methods") {
+		authenticationMethods := expandStringList(d.Get("authentication_methods").(*schema.Set).List())
+		authenticationMethodsValues := make([]sdk.AuthenticationMethods, len(authenticationMethods))
+		for i, v := range authenticationMethods {
+			authenticationMethodsValues[i] = sdk.AuthenticationMethods{Method: sdk.AuthenticationMethodsOption(v)}
+		}
+
+		baseReq := sdk.NewAlterAuthenticationPolicyRequest(id)
+		if len(authenticationMethodsValues) == 0 {
+			unsetReq := sdk.NewAuthenticationPolicyUnsetRequest().WithAuthenticationMethods(*sdk.Bool(true))
+			baseReq.WithUnset(*unsetReq)
+		} else {
+			setReq := sdk.NewAuthenticationPolicySetRequest().WithAuthenticationMethods(authenticationMethodsValues)
+			baseReq.WithSet(*setReq)
+		}
+
+		if err := client.AuthenticationPolicies.Alter(ctx, baseReq); err != nil {
```

Comment:
Instead of calling ALTER for each changed fields, we should modify set/unset requests and do one query at the end. Check resource_monitor.go

Reply:
done, please check